crates/polars-sql/src/functions.rs

```diff
@@ -969,9 +974,15 @@ impl SQLFunctionVisitor<'_> {
             ArrayMin => self.visit_unary(|e| e.list().min()),
             ArrayReverse => self.visit_unary(|e| e.list().reverse()),
             ArraySum => self.visit_unary(|e| e.list().sum()),
-            ArrayToString => self.try_visit_binary(|e, s| {
-                Ok(e.list().join(s))
-            }),
+            ArrayToString => match function.args.len(){
+                2 => self.try_visit_binary(|e, s| {
+                        Ok(e.list().join(s, true))
+                    }),
+                3 => self.try_visit_ternary(|e, s, ignore_nulls| {
+                    Ok(e.list().join(s, ignore_nulls))
+                }),
+                _ => polars_bail!(InvalidOperation: "Invalid number of arguments for ArrayToString: {}", function.args.len()),
+            }
```

Comment:
Sure, no problem ✌️

Reply:
Ok, need slightly different behaviour:

* 2-argument: `array_to_string(arr, sep)` needs ignore_nulls=True (by default).
* 3-argument: `array_to_string(arr, sep, null_value)` needs `fill_null(null_value)` first, _then_ concat.

If you like I can make those changes? Or happy to leave it in your capable hands ;)

Reply:
> Personally, It would be better for us to unify the default behavior of these functions(str.concat, list.jon, concat_str) to ignore null values. But might it has a different requirement for SQL stuff, I don't know...

Our own functions should definitely all behave consistently when called from Polars directly (eg: against DataFrame/Series); when called from SQL, they should behave as per SQL expectation, so both CONCAT and CONCAT_WS should set it to `true` - doing so would be considered a fix (for SQL) rather than a breaking change, as currently they do the wrong thing with NULL values 😅👍

Reply:
Bit later than planned, but done! `array_tostring` behaviour now conforms to SQL norms 👌